components/Service-Management.tsx

```diff
@@ -0,0 +1,142 @@
+// components/service-management.tsx
+'use client'
+
+import React, { useState } from 'react';
+import { useQuery, useMutation } from 'convex/react';
+import { api } from '@/convex/_generated/api';
+import { Button } from '@/components/ui/button';
+import { Input } from '@/components/ui/input';
+import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
+import { Dialog, DialogContent, DialogHeader, DialogTitle } from '@/components/ui/dialog';
+import { toast } from '@/components/ui/use-toast';
+import { Spinner } from '@/components/ui/spinner';
+import { Service } from '@/types/service';
+
+export const ServiceManagement: React.FC = () => {
+    const [isAddingService, setIsAddingService] = useState(false);
+    const [editingService, setEditingService] = useState<Service | null>(null);
+    const [newService, setNewService] = useState<Partial<Service>>({});
+
+    const services = useQuery(api.services.list);
+    const addService = useMutation(api.services.add);
+    const updateService = useMutation(api.services.update);
+    const deleteService = useMutation(api.services.remove);
+
+    const handleAddService = async () => {
+        try {
+            await addService(newService);

```

Comment:
**Suggestion:** Consider adding error handling for the case where `newService` is incomplete or invalid when attempting to add a new service. This will prevent sending incomplete data to the server and ensure that all required fields are filled. [enhancement, importance: 8]
```suggestion
            if (validateNewService(newService)) {
              await addService(newService);
            } else {
              toast({ title: "Error", description: "Please fill all required fields", variant: "destructive" });
            }
```